sklearn/ensemble/gradient_boosting.py

```diff
@@ -1234,6 +1235,7 @@ def feature_importances_(self):
             total_sum += stage_sum
 
         importances = total_sum / len(self.estimators_)
+        importances /= importances.sum()
```

Comment:
Please clarify how this relates to the new parameter

Reply:
The previous behavior normalized by the sum of importances, but did this on a per-tree basis before adding up all of the importances across trees.  The new parameter stops the normalization on the per-tree basis so it is added back in here.

The general idea is to switch from a `normalize then sum` workflow to a `sum then normalize` in order to avoid overweighting features that contribute less to total gini importance.

Reply:
But when are we using `normalize_feature_importances` actually. It seems that it is never used.

Reply:
It's never being used in GBM, but it is set `True` (by default) when used by Random Forests.

Reply:
I got that but I don't see a statement:

```python
if normalize_feature_importances:
    importance /= ...
```

Reply:
Ahhh, gotcha.  Ok, I'll add that in.  Thanks!

Reply:
Actually -- sorry, I think I misunderstood.  

It's only being used in the call to instantiate the `DecisionTreeRegressor` without adding in a totally separate class.

Reply:
Oh sorry I see. The computation of the importance is in the cython file, right.

Reply:
Right, so for GBMs only, we skip the normalization part of the computation in the cython file and then do the normalization here (which is reasonably cheap since it's just the single numpy operation no matter the number of trees)